### ISSUE TITLE
Crash: Media upload error results in a crash

### ISSUE THREAD

**Issue by rachelmcr** [wordpress-mobile/WordPress-iOS#4985]
### Expected behavior
When I try to upload media in the editor, and there is an error, I expect to receive an error message. For example, in the latest stable release, the app displays a "Failed" message in the post body or an error dialog when I try to upload a featured image:

![2016-03-19 15 13 30](https://cloud.githubusercontent.com/assets/8658164/13936233/29e3b7c8-efb4-11e5-820f-c097e8a6ffff.png)

### Actual behavior
When I try to upload media in the editor in the current beta, and my site returns the same error (a 403 error), the app crashes.

### Steps to reproduce the behavior

1. On a site that doesn't allow media uploads via the app, start a new blog post
2. Select the add media icon and select an image to upload from the device
3. Wait for the upload to finish — near the end of the upload progress, it will crash

Crash details:
```
Application Specific Information:
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFNumber isEqualToString:]: unrecognized selector sent to instance 0xb000000000001933'

Last Exception Backtrace:
0   CoreFoundation                       0x00000001816e1900 __exceptionPreprocess + 124
1   libobjc.A.dylib                      0x0000000180d4ff80 objc_exception_throw + 52
2   CoreFoundation                       0x00000001816e861c -[NSObject(NSObject) doesNotRecognizeSelector:] + 208
3   CoreFoundation                       0x00000001816e55b8 ___forwarding___ + 868
4   CoreFoundation                       0x00000001815e968c _CF_forwarding_prep_0 + 88
5   WordPress                            0x0000000100136f40 __67-[WordPressComApi HTTPRequestOperationWithRequest:success:failure:]_block_invoke (WordPressComApi.m:113)
6   AFNetworking                         0x00000001008f2844 __64-[AFHTTPRequestOperation setCompletionBlockWithSuccess:failure:]_block_invoke31 (AFHTTPRequestOperation.m:131)
7   libdispatch.dylib                    0x0000000181135630 _dispatch_call_block_and_release + 20
8   libdispatch.dylib                    0x00000001811355f0 _dispatch_client_callout + 12
9   libdispatch.dylib                    0x000000018113acf8 _dispatch_main_queue_callback_4CF + 1840
10  CoreFoundation                       0x0000000181698bb0 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 8
11  CoreFoundation                       0x0000000181696a18 __CFRunLoopRun + 1624
12  CoreFoundation                       0x00000001815c5680 CFRunLoopRunSpecific + 380
13  GraphicsServices                     0x0000000182ad4088 GSEventRunModal + 176
14  UIKit                                0x000000018643cd90 UIApplicationMain + 200
15  WordPress                            0x000000010007bb78 main (main.m:5)
16  ???                                  0x00000001811668b8 0x0 + 0
```

Related line in code: https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/WordPressApi/WordPressComApi.m#L113

##### Tested on iPhone 6, iOS 9.2.1, app version 6.1.0.20160318

**Comment by sendhil** [comment]
cc @SergioEstevao

**Comment by SergioEstevao** [comment]
I'm on it.

**Comment by SergioEstevao** [comment]
@rachelmcr is this on a self-hosted site? or wp.com? It looks wp.com but just want to be sure.

Are you trying to do it with what kind of user?

**Comment by rachelmcr** [comment]
This happened to me on a self-hosted Jetpack site. I'm logged in to the wp.com account used to make the Jetpack connection. (There's something wrong with the site to cause the error during upload.)

**Comment by rachelmcr** [comment]
I confirmed the fix in the latest beta release 6.1.0.20160331. Closing as resolved.
